Re-enable note ordering in DagRunModelView

uranusjr opened the pull request:
Following up https://github.com/apache/airflow/issues/30041 and #30043. This is actually much easier than I thought; instead of the associated proxy, we can reference the actual field here, and use a custom label.

uranusjr committed:
Re-enable note ordering in DagRunModelView

Instead of the associated proxy, we can reference the actual field here,
and use a custom label.

hussein-awala reviewed (changes requested):
Nice! the sort works, but there is a new problem because of this change.

When I delete a dag run, I got an error and here is the webserver log:
```
[2023-03-28T10:32:54.989+0000] {interface.py:790} ERROR - Delete record error: Dependency rule tried to blank-out primary key column 'dag_run_note.dag_run_id' on instance '<DagRunNote at 0x7f7189b93e10>' 
```
Can you check it?

uranusjr committed:
Properly set cascade on dag_run_note for deletion

This should allow a DAG run to be correctly deleted from web UI and
clean up its associated note row automatically.

ephraimbuddy commented on airflow/models/dagrun.py:
```diff
@@ -173,7 +173,12 @@ class DagRun(Base, LoggingMixin):
-    dag_run_note = relationship("DagRunNote", back_populates="dag_run", uselist=False)
+    dag_run_note = relationship(
+        "DagRunNote",
+        cascade="save-update, merge, delete, delete-orphan",
```
```suggestion
        cascade="all, delete-orphan",
```
`all` means `save-update, merge, refresh-expire, expunge, delete`. I think that's what we need

ephraimbuddy commented on airflow/www/views.py:
```diff
@@ -5036,7 +5036,7 @@ class DagRunModelView(AirflowPrivilegeVerifierModelView):
         "start_date",
         "end_date",
-        "note",
+        "dag_run_note.content",
```
`note` is a proxy to `dag_run_note.content`. Any reason for changing it?

uranusjr replied:
This is to allow the entry in `label_columns`. FAB auto-assigns an illegible label for a proxy, and doesn’t let me override it (or at least I couldn’t find a way to).

uranusjr committed:
Use 'all' cascade shorthand

Co-authored-by: Ephraim Anierobi <splendidzigy24@gmail.com>
